backend/Bindings/Edna.Bindings.User/UsersClient.cs

```diff
@@ -0,0 +1,36 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Edna.Utils.Http;
+using Newtonsoft.Json;
+
+namespace Edna.Bindings.User
+{
+    public class UsersClient
+    {
+        private readonly string _baseServiceUriString;
+        private readonly IHttpClientFactory _httpClientFactory;
+
+        internal UsersClient(string serviceUrlEnvironmentVariable, IHttpClientFactory httpClientFactory)
+        {
+            _httpClientFactory = httpClientFactory;
+            _baseServiceUriString = Environment.GetEnvironmentVariable(serviceUrlEnvironmentVariable);
+            if (string.IsNullOrEmpty(_baseServiceUriString))
+                throw new ArgumentException("User service URL was not provided.");
+        }
+
+        public async Task<Models.User[]> GetAllUsers(string assignmentId)
+        {
+            using HttpClient httpClient = _httpClientFactory.CreateClient(EdnaHttpHandler.Name);
+            httpClient.BaseAddress = new Uri(_baseServiceUriString);
+            string allUsersJson = await httpClient.GetStringAsync($"assignments/{assignmentId}/users");
```

Comment:
What happens if this API fails, let's say return an `http500` or `http204`?

Reply:
I've handled this from where this API call is being made, i.e. from inside the POST call functions.